### ISSUE TITLE
Installation fails due to 'XML' not available

### ISSUE THREAD

**Issue by LeonKalderon** [OHDSI/PatientLevelPrediction#175]
**Describe the bug**
Hello, 

When I try to install the packages with devtools I get the following error:

```
Downloading GitHub repo OHDSI/FeatureExtraction@master
Skipping 1 packages not available: XML
Installing 23 packages: DatabaseConnector, Andromeda, dplyr, rJava, SqlRender, ParallelLogger, readr, triebeard, generics, tidyselect, bit64, blob, DBI, plogr, bit, DatabaseConnectorJars, urltools, RSQLite, zip, dbplyr, snow, XML, hms
Installing packages into ‘/usr/local/lib/R/site-library’
(as ‘lib’ is unspecified)
Error: Failed to install 'FeatureExtraction' from GitHub:
  (converted from warning) package ‘XML’ is not available (for R version 3.5.1)
```

Many thanks for contributing to this great project. I'm am new to the community and I'm trying to set up PatientLevelPrediction.

I had managed to install everything and make it work (apart from the algorithms that are based on python packages) about ~ 10 days ago.

Before two days, I tried to make a clean installation and I'm completely stack. 
Sorry if this is not a bug and its just me.

**Set up (please run in R "sessionInfo()" and copy the output here):**
R version 3.5.1 (2018-07-02)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Debian GNU/Linux 9 (stretch)

Matrix products: default
BLAS: /usr/lib/openblas-base/libblas.so.3
LAPACK: /usr/lib/libopenblasp-r0.2.19.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C               LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8     LC_MONETARY=en_US.UTF-8   
 [6] LC_MESSAGES=C              LC_PAPER=en_US.UTF-8       LC_NAME=C                  LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] devtools_2.3.0 usethis_1.6.1 

loaded via a namespace (and not attached):
 [1] rstudioapi_0.11   magrittr_1.5      pkgload_1.1.0     R6_2.4.1          rlang_0.4.6       fansi_0.4.1       tools_3.5.1       pkgbuild_1.0.8   
 [9] sessioninfo_1.1.1 cli_2.0.2         withr_2.2.0       ellipsis_0.3.1    remotes_2.1.1     assertthat_0.2.1  digest_0.6.25     rprojroot_1.3-2  
[17] crayon_1.3.4      processx_3.4.3    callr_3.4.3       fs_1.4.2          ps_1.3.3          curl_4.3          testthat_2.3.2    memoise_1.1.0    
[25] glue_1.4.1        compiler_3.5.1    desc_1.2.0        backports_1.1.8   prettyunits_1.1.1

**To Reproduce**
install.packages('devtools')
library(devtools)
devtools::install_github("OHDSI/FeatureExtraction")
devtools::install_github('ohdsi/PatientLevelPrediction')

**Comment by jreps** [comment]
Hi Leon,

It seems to be an issue with 'FeatureExtraction' having 'XML' as a dependency.  I took a look a 'XML' here https://cran.r-project.org/web/packages/XML/index.html and saw that it annoyingly requires R 4.0.0 or greater (you have R version 3.5.1 (2018-07-02)).  This means you either need to: a) install an older version of XML (https://cran.r-project.org/src/contrib/Archive/XML/ and pick a version close to your R version date, maybe early 2019) or b) update R to 4.0.0 (this might be a good idea as your R version is a bit old now and you might find lots of packages becoming unavailable - but updating means re-installing all the packages so can be slow).

Best wishes,
Jenna

**Comment by jreps** [comment]
Also - for the python stuff, reticulate will automatically install miniConda and I have some R scripts to install the python dependancies (see section 4 https://github.com/OHDSI/PatientLevelPrediction/blob/master/inst/doc/InstallationGuide.pdf)

**Comment by LeonKalderon** [comment]
Hi @jreps ,

Thank you for your quick reply. 

That was really naive mistake from my side. I still can't understand why did it work at first place.

About python, I have already followed the installation guide but I get some errors related with python.

For example:

- Random Forest
`2020-07-09 13:59:13	[Main thread]	TRACE	PatientLevelPrediction	3	Error with runPlp:Error in value[[3L]](cond): Error: ImportError: cannot import name 'joblib'  Detailed traceback:    File "<string>", line 22, in <module> ` 
- Decision Tree
`2020-07-09 14:00:17	[Main thread]	TRACE	PatientLevelPrediction	3	Error with runPlp:Error in value[[3L]](cond): Error: ModuleNotFoundError: No module named 'sklearn.externals.joblib'  Detailed traceback:    File "<string>", line 20, in <module> `

These logs came from logs.txt file.

**Comment by jreps** [comment]
Ive been getting a warning about them depreciating joblibs from sklearn and ignoring it for a while - guess it finally caught up.  I'll update into the development branch ASAP and let you known once it is fixed.

R is often annoying, my guess is that XML happened to update recently and make the R 4.0.0 requirement between your installs.  Updating things with R is never as simple as it should be :)

**Comment by LeonKalderon** [comment]
Thanks a lot,

It's a good opportunity for me as well to dive into the details of this package and explore more the dependencies. :)

**Comment by jreps** [comment]
joblib is now a separate package. I've updated the python code to 'import joblib' (which will hopefully fix the error).  This means you need to have the joblib python package installed, I updated the python setup code but for you it will probably be easier to just run:
reticulate::conda_install(envname='r-reticulate', packages = 'joblib', forge = TRUE, pip = FALSE, pip_ignore_installed = TRUE, conda = "auto")

To install the development branch: devtools::install_github('ohdsi/PatientLevelPrediction', ref = 'development')

You may have to exit R and open a new session to get the updated version after the install - make sure 4.0.3 is loaded.

Hopefully the error will go away after that

**Comment by LeonKalderon** [comment]
Awesome thanks!!

**Comment by ted9219** [comment]
Hi Jenna, I just figure out a minor bug about joblib, so please consider this also. 

Bug) Cannot create models using the python (random forest, ada boost ...)
It seems like just skipped without model results.
I figure out that the sklearn.externals.joblib was deprecated.

env: PatientLevelPrediction version 4.0.2

In all python code in plp packages:
import sklearn.externals.joblib  -> import joblib
from sklearn.externals.joblib import Memory -> from joblib import Memory

after that python models were normally created.

**Comment by jreps** [comment]
Hi Chungsoo - great minds think alike, I edited the development branch with this changed yesterday - I need to move it into the main branch but want to do some more testing first :)

**Comment by LeonKalderon** [comment]
Hi Jenna,

Just a small update for the fix you made yesterday.

Everything worked fine after I installed joblibs manually. 
(`reticulate::conda_install(envname='r-reticulate', packages = 'joblib', forge = TRUE, pip = FALSE, pip_ignore_installed = TRUE, conda = "auto")`)

However, I tried to create a clean environment to test the automated installation of python throught PatientLevelPrediction package (`devtools::install_github('ohdsi/PatientLevelPrediction', ref = 'development')`) and this time there is a problem with `sklearn` package.

I got the following log in log.txt file when I run a python model:
`2020-07-10 13:19:36	[Main thread]	TRACE	PatientLevelPrediction	3	Error with runPlp:Error in value[[3L]](cond): Error in py_run_file_impl(file, local, convert): ModuleNotFoundError: No module named 'sklearn'  Detailed traceback:    File "<string>", line 16, in <module> `

The problem was fixed when I installed sklearn manually. (`reticulate::conda_install(envname='r-reticulate', packages = 'scikit-learn', forge = TRUE, pip = FALSE, pip_ignore_installed = TRUE, conda = "auto")
`) and everything works fine. 

I'm just reporting this bug in case you'd like to include the fix on the same PR.

Thanks :)
